fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSet.java

```diff
@@ -495,6 +495,12 @@ public static boolean isCastMatchAllowed(Function desc, Function candicate) {
                 return false;
             }
         }
+        if (FunctionCallExpr.STRING_SEARCH_FUNCTION_SET.contains(desc.functionName())) {
+            if (descArgTypes[1].isStringType() && candicateArgTypes[1].isArrayType()) {
+                // The needles arg of search functions should not be allowed to cast from string.
```

Comment:
why this is not allowed?

Reply:
@zhiqiang-hhhh That's correct. see testcase.

Reply:
This is to prohibit casting the needles argument from string ->array<text>

Reply:
why cast sting to array<text> leading to core?

Reply:
because the cast function wrapped a nullable layer above the result type when it encountered a string input.